File: src/main/java/com/javarush/island/shevchenko/entities/animals/herbivores/Buffalo.java
package com.javarush.island.shevchenko.entities.animals.herbivores;

import com.javarush.island.shevchenko.entities.organism.Animal;
import com.javarush.island.shevchenko.entities.organism.interfaces.*;
import com.javarush.island.shevchenko.settings.IslandSettings;
import com.javarush.island.shevchenko.util.Randomizer;


/**
 * Класс Buffalo представляет существо буйвола на острове животных.
 * Буйволы являются травоядными и используют поведение из интерфейса Herbivorous для питания.
 */
public class Buffalo extends Animal implements Herbivorous {

    public Buffalo() {
        this.aClass = Buffalo.class;
        this.weight = Randomizer.getRandom(IslandSettings.ANIMAL_PARAMETERS.get(Buffalo.class)[0] / 2, IslandSettings.ANIMAL_PARAMETERS.get(Buffalo.class)[0]);
    }
}

File: src/main/kotlin/ru/cooper/island/entities/animals/herbivores/Buffalo.kt
package ru.cooper.island.entities.animals.herbivores

import ru.cooper.island.entities.animals.herbivores.function.initAnimal
import ru.cooper.island.entities.organism.Animal
import ru.cooper.island.entities.organism.interfaces.Herbivorous

class Buffalo : Animal(), Herbivorous {
    init { initAnimal<Buffalo>() }
}
